### PR TITLE
Add permissions check to block diagram API

### PR DESCRIPTION
**Pull request by cabarnes**
* Requires the user to be logged in to access the block diagram API.
* User can only access the user's block diagrams.
* Any superuser accounts can access all block diagrams through the API.
* Sets user to the logged in user that created the block diagram.

### CODE REVIEW

**Comment by aninternetof** [comment]
What do you think of these tweaks?

* Requires the user to be logged in to access the block diagram API.
* _Any user can **read** all users' block diagrams_
* _User can **write** only his own block diagrams._
* Any superuser accounts can read and write all block diagrams through the API.
* Sets user to the logged in user that created the block diagram.

Being able to read all other users' block diagrams will make it easier for us in the future to make the "clone" function, where a user makes his own copy of somebody else's block diagram.

Thoughts?

**Comment by cabarnes** [comment]
@aninternetof So, would all block diagrams be publicly viewable or would the user have control over which are shared?

**Comment by aninternetof** [comment]
I vote for all block diagrams being publicly viewable. I believe that's what Scratch does -- encourage (force) sharing.

**Comment by cabarnes** [comment]
Then they can buy premium to have `private` block diagrams ;)

File: mission_control/views.py [hunk 1]
```diff
@@ -43,5 +43,18 @@ class BlockDiagramViewSet(viewsets.ModelViewSet):
...
+    permission_classes = (permissions.IsAuthenticated, )
     filter_backends = (DjangoFilterBackend,)
     filter_fields = ('user',)
+
+    def perform_create(self, serializer):
+        """Perform the create operation."""
+        user = self.request.user
+        serializer.save(user=user)
+
+    def perform_update(self, serializer):
```
Comment (aninternetof) [inline mission_control/views.py]: Does this let a superuser update anyone's block diagram?

Comment (cabarnes) [inline mission_control/views.py]: No, it won't. I wasn't sure if there would really be a use for it. I could change the `if` statement to also check for superuser if so.

Comment (aninternetof) [inline mission_control/views.py]: Sounds good -- let's leave it out.

**Pull request merged**